src/include/miopen/multimarginloss/problem_description.hpp

```diff
@@ -0,0 +1,133 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include "miopen/miopen.h"
+#include <miopen/activ.hpp>
+#include <miopen/problem_description_base.hpp>
+#include <miopen/tensor.hpp>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace multimarginloss {
+
+struct ForwardProblemDescription : ProblemDescriptionBase
+{
+    ForwardProblemDescription(const TensorDescriptor& iDesc_,
+                              const TensorDescriptor& tDesc_,
+                              const TensorDescriptor& wDesc_,
+                              const TensorDescriptor& oDesc_,
+                              const long p_,
+                              const float margin_,
+                              const float divisor_)
+        : iDesc(iDesc_),
+          tDesc(tDesc_),
+          wDesc(wDesc_),
+          oDesc(oDesc_),
+          p(p_),
+          margin(margin_),
+          divisor(divisor_)
+    {
+        if(iDesc.GetType() != oDesc.GetType() || iDesc.GetType() != wDesc.GetType())
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "MultiMarginLoss: Input, output, weight tensor types do not match.");
+        }
+        if(tDesc.GetType() != miopenInt64)
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "MultiMarginLoss: Target tensor type should be miopenInt64.");
+        }
+        if(iDesc.GetNumDims() != 2)
+        {
+            MIOPEN_THROW(miopenStatusBadParm, "MultiMarginLoss: Input tensor need to be 2D tensor");
```

Comment:
I know it but in Modnn source code in [here](https://github.com/moreh-dev/backend/blob/0828817cd9f7e3f0832790eced825d68dfecbfef/modnn/core/op/loss.cpp#L3540) for this op they verified the input tensor to be 2D. Let's leave it for AMD reviewers when we push this PR upstream

Reply:
CHECK_DIM_LIMIT here checks whether dim number of input tensor <= 2, not exactly equal 2

Reply:
Ok because the MultilabelSoftmarginLoss operation I implemented before also used CHECK_DIM_LIMIT and in pytorch doc for that op the input tensor can only be 2D so I misunderstand the purpose of this macro. But actually modnn didn't work with 1D case, there is a condition to check input tensor is 2D in IR and I read modnn source code for this op, it definitely cannot work with 1D tensor. For my MIOpen code, modify the code to support for 1D required to change a lot of code unfortunately. And I found that not necessary because: (1) People can easily pass a 2D tensor (1,C) to this op and it will work like a 1D tensor. (2) N = 1 is rarely used because this kernel is parallel by N

Reply:
ok 1D cases are just for user convenient so let pass it now